site/docs/hosting/vps.md

```diff
@@ -45,131 +120,486 @@ cd /etc/systemd/system
 2. Open your new service file with an editor:
 
 ```sh
-nano bot1.service
+nano <app-name>.service
 ```
 
+replacing `<app-name>` with any identifier.
+
+> `<app-name>.service` will be the name of your service.
+
 3. Add the following content:
 
 ```text
+[Unit]
+After=network.target
+
 [Service]
-ExecStart=<start_command>
+Environment=BOT_TOKEN=<token>
+Type=simple
+User=<username>
+ExecStart=<start-command>
 Restart=on-failure
 
 [Install]
 WantedBy=multi-user.target
 ```
 
-> Replace `<start_command>` with the command you got above.
->
-> Also note that if Deno is installed for a different user other than root, you may need to specify it in the service section like `User=<the_user>`.
+replacing `<token>` with your bot's token, `<launch-command>` with the command you received [above](#getting-the-start-command), and `<username>` with the name of the user on whose behalf the app is being started.
+
 > For more information on the unit files, visit [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_basic_system_settings/assembly_working-with-systemd-unit-files_configuring-basic-system-settings).
 
-3. Reload systemd whenever you edit the service:
+4. Reload systemd whenever you edit the service:
 
-```sh
+```sh:no-line-numbers
 systemctl daemon-reload
 ```
 
 ### Managing the Service
 
-#### Start
+```sh:no-line-numbers
+# Replace `<service-name>` with the file name of the service you created.
 
-```sh
-systemctl start <service_name>
+# To start the service
+systemctl start <service-name>
+
+# To view the service logs
+systemctl status <service-name>
+
+# To restart the service
+systemctl restart <service-name>
+
+# To stop the service
+systemctl stop <service-name>
+
+# To enable the service to start when the server boots
+systemctl enable <service-name>
+
+# To disable service startup when the server boots
+systemctl disable <service-name>
 ```
 
-> Replace `<service_name>` with the file name of the service.
-> Example: `systemctl start bot1`
+### PM2
 
-#### Run on Boot
+[PM2](https://pm2.keymetrics.io/) is a daemon process manager for Node.js that will help you manage and keep your app online 24/7.
+
+> PM2 is designed specifically to run applications written in Node.js.
+> However, it can also be used to run applications written in other languages or for other frameworks.
+
+### Installing
 
 ```sh
-systemctl enable <service_name>
+npm install pm2@latest -g
+
+# If using Yarn
+yarn global add pm2
```

Comment:
Consider using a code environment if you want to branch out for different package managers.

Reply:
I did not change this part in any way. It is identical to the previous version of the article.

Reply:
That's alright, but while we're at it, why not improve it? :)

Reply:
Yes, we can improve it. However, I need to understand what you mean by "code environment"?

Reply:
This is called a "code group"

Reply:
Even the corresponding html tag is called that

Reply:
Should we add a command for pnpm as well, as is done in [other manuals](https://grammy.dev/guide/getting-started.html)?

Reply:
Yep!